packages/gui/src/components/nfts/provider/hooks/useNFTData.ts

```diff
@@ -0,0 +1,295 @@
+import { EventEmitter } from 'events';
+
+import { type NFTInfo, type Wallet } from '@chia-network/api';
+import {
+  useLazyGetNFTsCountQuery,
+  useLazyGetNFTsQuery,
+  useGetNFTWallets,
+  useGetLoggedInFingerprintQuery,
+} from '@chia-network/api-react';
+import debug from 'debug';
+import { useState, useCallback, useEffect, useMemo, useRef } from 'react';
+
+import type NFTState from '../../../../@types/NFTState';
+import useNFTCoinEvents from '../../../../hooks/useNFTCoinEvents';
+import useStateAbort from '../../../../hooks/useStateAbort';
+import getNFTId from '../../../../util/getNFTId';
+import limit from '../../../../util/limit';
+import { getChangedEventName } from './useNFTDataOnDemand';
+
+const log = debug('chia-gui:useNFTData');
+
+type UseNFTDataProps = {
+  pageSize?: number;
+  concurrency?: number;
+};
+
+// warning: only used by NFTProvider
+export default function useNFTData(props: UseNFTDataProps) {
+  const { pageSize = 24, concurrency = 10 } = props;
+
+  const abortControllerRef = useRef<AbortController>(new AbortController());
+  const [nfts] = useState(() => new Map<string, NFTInfo>());
+  const add /* immutable until concurrency change */ = useMemo(() => limit(concurrency), [concurrency]);
+
+  const events /* immutable */ = useMemo(() => {
+    const eventEmitter = new EventEmitter();
+    eventEmitter.setMaxListeners(Infinity);
```

Comment:
♾️ seems like a lot :). Will setting this make it more difficult to troubleshoot legitimate issues?

Reply:
I see Infinity used elsewhere. My primary concern is that if there's a problem with something registering too many listeners, it may be difficult to track down the issue.

Reply:
We have on one screen 50 nft card, each card can have 5+ useNFTMetadata hooks => 5x50 = 250 events.
Notifications screen are using useNFTMEtadata as well => 4 * 10 = ~ 50
NFTAutocomplete can use more.
We cam add limit but I do not know what is a good number. 
Maybe 1000 ?

Reply:
I'd be fine with a large number like 10_000. I don't think we need to hold this PR up for this change though.